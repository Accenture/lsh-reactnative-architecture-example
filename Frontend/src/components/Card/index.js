import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import styles from './Card.style';

const Card = props => (
  <View style={styles.containerStyle}>{props.children}</View>
);
Card.defaultProps = {
  children: PropTypes.element.isRequired,
};

Card.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};
export default Card;
