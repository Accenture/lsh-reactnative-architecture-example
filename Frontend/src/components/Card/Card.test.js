import React from 'react';
import { shallow } from 'enzyme';
import Card from './index';

describe('Tests card component', () => {
  it('matches snapshot', () => {
    const comp = shallow(<Card />);
    expect(comp).toMatchSnapshot();
  });
});
