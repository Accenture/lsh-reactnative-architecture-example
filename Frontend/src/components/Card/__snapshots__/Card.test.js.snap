// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests card component matches snapshot 1`] = `
<Component
  style={
    Object {
      "borderBottomWidth": 0,
      "borderColor": "#ddd",
      "borderRadius": 2,
      "borderWidth": 1,
      "elevation": 1,
      "marginLeft": 5,
      "marginRight": 5,
      "marginTop": 10,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 2,
    }
  }
>
  <Component />
</Component>
`;
